src/middlewared/middlewared/plugins/kubernetes_linux/lifecycle.py

```diff
@@ -0,0 +1,141 @@
+import asyncio
+import errno
+import json
+import os
+
+from middlewared.service import CallError, private, Service
+
+
+class KubernetesService(Service):
+
+    @private
+    async def post_start(self):
+        # TODO: Add support for migrations
+        async def node_online(middleware):
+            while not (await middleware.call('k8s.node.config'))['node_configured']:
+                await asyncio.sleep(2)
```

Comment:
This seems like an overengineering use of coroutines since its just a sleep, could very well be a while with a counter